lua/nvim-treesitter/install.lua

```diff
@@ -490,11 +490,36 @@ function M.uninstall(...)
       end
 
       local parser_lib = utils.join_path(install_dir, lang) .. ".so"
-
-      local command_list = {
-        shell.select_rm_file_cmd(parser_lib, "Uninstalling parser for " .. lang),
-      }
-      M.iter_cmd(command_list, 1, lang, "Treesitter parser for " .. lang .. " has been uninstalled")
+      local all_parsers = vim.api.nvim_get_runtime_file("parser/" .. lang .. ".so", true)
+      if vim.fn.filereadable(parser_lib) == 1 then
```

Comment:
we could also try to uninstall parser from all locations but I think `:TSUninstall` is rather thought for managing parsers installed nvim-treesitter and not for instance the built-in ones.

Reply:
I agree the installer should be managed for nvim-treesitter and not custom installs. Actually, we also could move the installer in a separate plugin, and provide a proper API to get the nvim-treesitter queries versions.
Let me know if this seems like a good idea, although it's still nice to have this in nvim-treesitter.

Reply:
Yes. Core will ship with some parsers soon(tm), and these should never be touched. Nvim-treesitter should only touch its own install location (but inform about others).